### PR TITLE
vsphere: Call DeleteGovmomiClient on Teardown

### LINKED ISSUE #614: vsphere: DeleteGovmomiClient on Teardown
Add cleanup to new Teardown callback

### PR DESCRIPTION
**Pull request by caavery**
Fixes #614

Signed-off-by: Cathy Avery <cavery@redhat.com>

### CODE REVIEW

File: pkg/adaptor/cloud/vsphere/provider.go [hunk 1]
```diff
@@ -295,5 +295,7 @@ func (p *vsphereProvider) DeleteInstance(ctx context.Context, instanceID string)
 }
 
 func (p *vsphereProvider) Teardown() error {
+	logger.Printf("Logout user %s", p.serviceConfig.UserName)
+	DeleteGovmomiClient(p.gclient)
```
Comment (wainersm) [inline pkg/adaptor/cloud/vsphere/provider.go:299]: Hi @caavery !

The `DeleteGovmomicClient()` function (https://github.com/caavery/cloud-api-adaptor/blob/939c2394bb17aef826a7207bdc1f552603c850c5/pkg/adaptor/cloud/vsphere/vsphere.go#L78) has any error logged but an error object is not return. Why?

I wondering if it wouldn't be important to raise any error through the `Teardown()`'s return.

Comment (caavery) [reply pkg/adaptor/cloud/vsphere/provider.go:299]: Yes I think you are right. Fixing it now.

Thanks!

**Review by yoheiueda — APPROVED** [review]
LGTM

**Review by bpradipt — APPROVED** [review]
/lgtm
Thanks @caavery

**Review by wainersm — APPROVED** [review]
@caavery thanks for addressing my comment. LGTM.

**Pull request merged**